packages/cojson-transport-ws/src/BatchedOutgoingMessages.ts

```diff
@@ -1,48 +1,152 @@
-import type { SyncMessage } from "cojson";
+import type { DisconnectedError, SyncMessage } from "cojson";
+import type { Peer } from "cojson";
+import {
+  type CojsonInternalTypes,
+  PriorityBasedMessageQueue,
+  cojsonInternals,
+  logger,
+} from "cojson";
 import { addMessageToBacklog } from "./serialization.js";
+import type { AnyWebSocket } from "./types.js";
+import {
+  hasWebSocketTooMuchBufferedData,
+  isWebSocketOpen,
+  waitForWebSocketBufferedAmount,
+  waitForWebSocketOpen,
+} from "./utils.js";
+
+const { CO_VALUE_PRIORITY } = cojsonInternals;
 
 export const MAX_OUTGOING_MESSAGES_CHUNK_BYTES = 25_000;
 
-export class BatchedOutgoingMessages {
+export class BatchedOutgoingMessages
```

Comment:
Moved the PriorityBasedMessageQueue inside this class, so now outgoing messages prioritization is done here.

Now we can fully control the outgoing messages scheduling from this class, and I've changed to scheduling from debouncing to a simple delay of 10ms before start the messages processing.

This makes the batching and prioritization more effective as subsequent operations are more likely to be synced and reduces its overhead because we don't do anymore setTimeout/clearTimeout on each outgoing.push.